pkg/backup/remap_crd_version_action.go

```diff
@@ -61,6 +61,7 @@ func (a *RemapCRDVersionAction) Execute(item runtime.Unstructured, backup *v1.Ba
 	}
 
 	// We've got a v1 CRD, so proceed.
+	// TODO: why is this causing int/float errors?
 	var crd apiextv1.CustomResourceDefinition
 	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(item.UnstructuredContent(), &crd); err != nil {
```

Comment:
Changing this to a JSON marshalling dance causes the unit test to pass, but I have no measured impact on actual restores yet.

```go
	var crd apiextv1.CustomResourceDefinition
	js, err := json.Marshal(item.UnstructuredContent())
	if err != nil {
		return nil, nil, errors.Wrap(err, "unable to convert unstructured item to JSON")
	}

	if err = json.Unmarshal(js, &crd); err != nil {
		return nil, nil, errors.Wrap(err, "unable to convert JSON to CRD")
	}
```
I'm also exploring using the helpers in the `unstructured` package, but I think they may be unwieldy given the number of fields we access in this plugin.

I'll try to see if I can log any sort of timing data to help make this decision.

Reply:
Based on the timing data at https://docs.google.com/spreadsheets/d/16YNPgBOlAzSBxyg_6Bsm0cLtZ2hBob0Vr_nz9vwYYL8/edit?usp=sharing in the `crd-log` tab, I'm leaning towards doing the JSON marshalling dance and no the unstructured helpers & casting for readability. Thoughts, @skriss and @ashish-amarnath?

Reply:
So basically, using `FromUnstructured` does not properly convert the integer value to a float, but marshaling the Unstructured to JSON and then using `json.Unmarshal` does? Is that correct?

Reply:
Yep, that appears to be the case. Also in my test case, if the value was `5.0`, it would fail with `FromUnstructured`, but it would be fine with `5.2`. So any whole number appears to have problems with the `FromUnstructured` implementation right now.

Reply:
OK. I'm onboard with going through JSON for this -- let's just add a good comment explaining.

Reply:
If you wanted to do the same in the other spot too, that'd be OK - not sure if it's worth changing or not, up to you.

Reply:
The problem sounds super weird! But the changes seem fine. 
The impact of the change doesn't seem to be that much. This may be one exception to keep commented code. WDYT?

Reply:
@ashish-amarnath I'll keep a mention of the full path to the call, but not all the error handling, as well as a mention of the Kubernetes issue.

@skriss I'll look at the other site and see if it's worth doing the JSON marshalling. If it's a single field, it's probably not worth it.